src/components/helpers/encoders.py

```diff
@@ -96,30 +96,68 @@ def encode_jpeg_bytes(arr: NDArray, is_depth: bool = False) -> bytes:
 def encode_pcd(arr: NDArray):
     """
     Encodes numpy points data into bytes decodable in the PCD format.
+    If color information is available (6 columns: x, y, z, r, g, b), it packs the color into a single float field.
 
     Args:
-        arr (NDArray): points data
-
+        arr (NDArray): points data (shape Nx3 for uncolored or Nx6 for colored)
     Returns:
-        bytes: PCD bytes
+        Tuple[bytes, CameraMimeType]: PCD bytes and associated mime type
     """
-    # DepthAI examples indicate that we need to normalize data by / 1000
-    # https://github.com/luxonis/depthai/blob/f4a0d3d4364565faacf3ce9f131a42b2b951ec1b/depthai_sdk/src/depthai_sdk/visualize/visualizers/viewer_visualizer.py#L72
-    flat_array = arr.reshape(-1, arr.shape[-1]) / 1000.0
-    version = "VERSION .7\n"
-    fields = "FIELDS x y z\n"
-    size = "SIZE 4 4 4\n"
-    type_of = "TYPE F F F\n"
-    count = "COUNT 1 1 1\n"
-    height = "HEIGHT 1\n"
-    viewpoint = "VIEWPOINT 0 0 0 1 0 0 0\n"
-    data = "DATA binary\n"
-    width = f"WIDTH {len(flat_array)}\n"
-    points = f"POINTS {len(flat_array)}\n"
-    header = f"{version}{fields}{size}{type_of}{count}{width}{height}{viewpoint}{points}{data}"
-    header_bytes = bytes(header, "UTF-8")
-    float_array = np.array(flat_array, dtype="f")
-    return (header_bytes + float_array.tobytes(), CameraMimeType.PCD)
+    if arr.shape[1] == 3:
+        # Uncolored pointcloud: convert coordinates from mm to m
+        points_xyz = arr[:, :3] / 1000.0
+        version = "VERSION .7\n"
+        fields = "FIELDS x y z\n"
+        size = "SIZE 4 4 4\n"
+        type_of = "TYPE F F F\n"
+        count = "COUNT 1 1 1\n"
+        height = "HEIGHT 1\n"
+        viewpoint = "VIEWPOINT 0 0 0 1 0 0 0\n"
+        width = f"WIDTH {points_xyz.shape[0]}\n"
+        points_count = f"POINTS {points_xyz.shape[0]}\n"
+        data = "DATA binary\n"
+        header = f"{version}{fields}{size}{type_of}{count}{width}{height}{viewpoint}{points_count}{data}"
+        header_bytes = bytes(header, "UTF-8")
+        float_array = np.array(points_xyz, dtype="f")
+        return (header_bytes + float_array.tobytes(), CameraMimeType.PCD)
+    elif arr.shape[1] == 6:
+        # Colored pointcloud: first three columns are coordinates (in mm), last three are color (assumed in [0,255])
+        points_xyz = arr[:, :3] / 1000.0  # convert to meters
```

Comment:
maybe go straight to float32 instead of during colored_points setter.

`points_xyz = (arr[:, :3] / 1000.0).astype(np.float32)`

Reply:
Done